packages/fhir-router/src/fhirrouter.ts

```diff
@@ -143,31 +183,43 @@ async function patchResource(req: FhirRequest, repo: FhirRepository): Promise<Fh
   return [allOk, resource];
 }
 
+export type RestInteraction =
+  | CapabilityStatementRestInteraction['code']
+  | CapabilityStatementRestResourceInteraction['code']
+  | 'execute';
+type RouteMetadata = {
+  interaction: RestInteraction;
+};
+
 export class FhirRouter extends EventTarget {
-  readonly router = new Router<FhirRouteHandler>();
+  readonly router = new Router<FhirRouteHandler, RouteMetadata>();
   readonly options: FhirOptions;
 
   constructor(options = {}) {
     super();
     this.options = options;
 
-    this.router.add('GET', '', searchMultipleTypes);
-    this.router.add('POST', '', batch);
-    this.router.add('GET', ':resourceType', search);
-    this.router.add('POST', ':resourceType/_search', searchByPost);
-    this.router.add('POST', ':resourceType', createResource);
-    this.router.add('GET', ':resourceType/:id', readResourceById);
-    this.router.add('GET', ':resourceType/:id/_history', readHistory);
-    this.router.add('GET', ':resourceType/:id/_history/:vid', readVersion);
-    this.router.add('PUT', ':resourceType', conditionalUpdate);
-    this.router.add('PUT', ':resourceType/:id', updateResource);
-    this.router.add('DELETE', ':resourceType/:id', deleteResource);
-    this.router.add('PATCH', ':resourceType/:id', patchResource);
-    this.router.add('POST', '$graphql', graphqlHandler);
+    this.router.add('GET', '', searchMultipleTypes, { interaction: 'search-system' });
```

Comment:
This is definitely starting to paint a clearer picture for the future.

Something like:
* New outer type such as `FhirProcessor` which wraps `FhirRouter` and `FhirRepository`
* `registerHandler(handler: OperationHandler)` which adds routes and whatnot

I'm not sure if FHIR has a good term that encompasses both `interaction` and `operation` ?